app/forms/nsm/steps/disbursement_cost_form.rb

```diff
@@ -19,19 +19,27 @@ def apply_vat
         @apply_vat.nil? ? record.vat_amount.to_f.positive? : @apply_vat == 'true'
       end
 
+      # rubocop:disable Metrics/MethodLength
       def calculation_rows
-        [
-          [translate(:before_vat), translate(:after_vat)],
-          [{
-            text: NumberTo.pounds(total_cost_pre_vat),
-            html_attributes: { id: 'total-without-vat' }
-          },
-           {
-             text: NumberTo.pounds(total_cost),
-             html_attributes: { id: 'total-with-vat' },
-           }],
-        ]
+        # this logic is needed to prevent invalid values erroring in the NumberTo methods
+        if total_cost_pre_vat.is_a?(String)
+          record.errors.add(:total_cost_without_vat, :not_a_decimal)
```

Comment:
I don't think we should be doing any `errors.add` stuff anywhere other than in validation methods. It's a weird side effect to add, and a bit unpredictable, because if we call `calculation_rows` after we've rendered the relevant bits of the form it will have no effect, but if we call it beforehand it won't. Plus this error should be added by the existing validation, right?